### ISSUE TITLE
Run pretraining of llama 7b/8b models on GPUs using a3/1vm.sh

### ISSUE THREAD

**Issue by giacomoni** [AI-Hypercomputer/maxtext#2193]
### Bug report

It looks like there is no workflow to build a docker image for GPUs that allows to correctly run the MaxText/configs/a3/llama2_7b/1vm.sh script. 

I have been relying on the `docker_build_dependency_image.sh` and tried different combinations of input arguments:

1) DEVICE=gpu
2) MODE=pinned DEVICE=gpu
3) DEVICE=gpu MODE=stable_stack BASEIMAGE=us-central1-docker.pkg.dev/deeplearning-images/jax-ai-image/gpu:jax0.5.1-cuda_dl25.02-rev1

Each of the above fails at some point, wither during the building prodcess of the docker image or runtime.

1) It fails when trying to run the script, since the attention mask is of type float but transformer-engine's dot product attention expects a bool mask

2) It fails when building the image, because pip can't resolve the requirements and constraints at the same time

3) I get a RESOURCE_EXHAUSTED jax compilation error

I also fixed the issues in 1) (patched the code) and I eventually got the script to run correctly, until JAX compilation breaks due to RESOURCE_EXHAUSTED. 

I am running the code on 1 node with 8 H100, so the current configs for llama2 7b and llama3 8b should run. 

I have also tried all the above not from master but the latest available tag (i.e. `tpu-recipes-v0.1.4`).

Any suggestion on how to get a working and stable Docker image for GPU execution?

Any suggestion

**Comment by giacomoni** [comment]
UPDATE: [JAX-Toolbox ](https://github.com/NVIDIA/JAX-Toolbox/tree/main)provides working containers out of the box for MaxText.

**Comment by RissyRan** [comment]
Hi @bvandermoon, could you help fix or triage the issue for GPU with docker image? Thanks!

**Comment by Skylion007** [comment]
Case 1) is fixed by: https://github.com/AI-Hypercomputer/maxtext/pull/2303

**Comment by Skylion007** [comment]
Okay, yeah confirmed the latest maxtext nightlys are broken sadly. :( and need #2303 . This was the case of the nvidia+jax+maxtext-2025-08-21 docker image at least.
